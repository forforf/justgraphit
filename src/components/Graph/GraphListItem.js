import React from 'react';
import {ListItem, ListItemIcon, ListItemText} from '@mui/material';


import DeleteIcon from '@mui/icons-material/Delete';

const GraphListItem = ({graphName, deleteGraph, handleClick}) => {
  return (
    <ListItem onClick={() => handleClick(graphName)}>
      <ListItemText primary={graphName} />
      <ListItemIcon>
        <DeleteIcon onClick={() => deleteGraph(graphName)} />
      </ListItemIcon>
    </ListItem>
  )
};

export default GraphListItem;
