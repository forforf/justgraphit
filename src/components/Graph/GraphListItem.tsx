import React from 'react';
import {ListItem, ListItemIcon, ListItemText} from '@mui/material';


import DeleteIcon from '@mui/icons-material/Delete';
import {GraphHandler, GraphName} from "../../JustGraphitTypes";

export type GraphListItemProps = {
  graphName: GraphName;
  deleteGraph: GraphHandler;
  handleClick: GraphHandler;
}
const GraphListItem = ({graphName, deleteGraph, handleClick}: GraphListItemProps): JSX.Element => {
  return (
    <ListItem onClick={() => handleClick(graphName)}>
      <ListItemText primary={graphName} />
      <ListItemIcon>
        <DeleteIcon onClick={() => deleteGraph(graphName)} />
      </ListItemIcon>
    </ListItem>
  )
};

export default GraphListItem;
