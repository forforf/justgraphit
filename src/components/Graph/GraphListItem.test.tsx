import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import GraphListItem, { GraphListItemProps } from './GraphListItem';

const graphName = 'graph name';
let deleteGraph = null;
let handleClick = null;
let graphListItemProps: GraphListItemProps;

describe('happy path', () => {
  beforeEach(() => {
    deleteGraph = jest.fn();
    handleClick = jest.fn();
    graphListItemProps = { graphName, deleteGraph, handleClick };
  });

  test('deep snapshot', () => {
    const component = renderer.create(
      <GraphListItem {...graphListItemProps} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('shallow snapshot', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<GraphListItem {...graphListItemProps} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
