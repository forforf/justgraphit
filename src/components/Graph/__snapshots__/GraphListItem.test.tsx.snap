// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy path deep snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-0.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-0.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
}

.emotion-1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-3 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-0"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="MuiListItemText-root emotion-1"
  >
    <span
      className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-2"
    >
      graph name
    </span>
  </div>
  <div
    className="MuiListItemIcon-root emotion-3"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
      data-testid="DeleteIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
  </div>
</li>
`;

exports[`happy path shallow snapshot 1`] = `
<ForwardRef(ListItem)
  onClick={[Function]}
>
  <ForwardRef(ListItemText)
    primary="graph name"
  />
  <ForwardRef(ListItemIcon)>
    <Memo
      onClick={[Function]}
    />
  </ForwardRef(ListItemIcon)>
</ForwardRef(ListItem)>
`;
